### PR TITLE
add C&C in eternum

### PR DESCRIPTION
**Pull request by FrostStarBook**
add C&C contract and Front-end integrated

### CODE REVIEW

File: sdk/packages/eternum/src/types/index.ts [hunk 1]
```diff
@@ -3,6 +3,13 @@ import { Account, num } from "starknet";
 interface SystemSigner {
   signer: Account;
 }
+export interface MintCC extends SystemSigner {
```
Comment (ponderingdemocritus) [inline sdk/packages/eternum/src/types/index.ts:6]: this doesn't need to exist - just use SystemSigner

File: sdk/packages/eternum/src/constants/contracts.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export const DEV_CONTRACTS = {
   TRAVEL_SYSTEMS: "0x70717be365c143d9f4ae207e420d0c3525a7c79197a20e6e63e4dec0b1b26cd",
   TEST_REALM_SYSTEMS: "0xee850f8d8ded18763d23e35d3592f9549081124bee0c32ffb2ef355deb1b68",
   TEST_RESOURCE_SYSTEMS: "0x17505387265fa24a4846cb3b3207aa92c3e6f918f74883a43be959265d19e44",
+  CC_CONTRACT_ADDRESS:"0x1e26e39c8d6d9d03179426d6a95190001b72a75645da0a3cff264c936bd3f1",
```
Comment (ponderingdemocritus) [inline sdk/packages/eternum/src/constants/contracts.ts:14]: why are production and development the same?

File: contracts/src/systems/cc_map/interface.cairo [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// ---------------------------------------- interface -------------------------------------
+#[starknet::interface]
+trait CCInterface<TContractState> {
```
Comment (ponderingdemocritus) [inline contracts/src/systems/cc_map/interface.cairo:3]: this should be exposed in the CC repo and imported, it shouldn't exist within eternum repo

File: contracts/src/systems/cc_map/contract.cairo [hunk 4]
```diff
@@ -0,0 +1,141 @@
+// ---------------------------------------- contract --------------------------------------
```
Comment (ponderingdemocritus) [inline contracts/src/systems/cc_map/contract.cairo:1]: this shouldn't exist in this repo either

File: sdk/packages/eternum/src/types/index.ts (hunk 1 above)
Comment (FrostStarBook) [reply sdk/packages/eternum/src/types/index.ts:6]: np

File: contracts/src/systems/cc_map/contract.cairo (hunk 4 above)
Comment (FrostStarBook) [reply contracts/src/systems/cc_map/contract.cairo]: This is dojo::contract. We are not able to import the package by dependency, we need to have an implementation in the repo.

File: sdk/packages/eternum/src/constants/contracts.ts (hunk 2 above)
Comment (FrostStarBook) [reply sdk/packages/eternum/src/constants/contracts.ts]: How to get Production Contract Address

**Comment by ponderingdemocritus** [comment]
I think this should exist on an alternatively hosted client - so we decentralise the clients

**Comment by ponderingdemocritus** [comment]
I have added instructions in the client readme - can you spin an alt client up?

**Comment by FrostStarBook** [comment]
> I have added instructions in the client readme - can you spin an alt client up?

I've seen it, it's written very clearly. We will remake one. The front end just needs a redirect button or something like that. The contract part will submit a PR.

**Comment by ponderingdemocritus** [comment]
It makes sense to maintain a fork of this rather than PR. This way the C&C client can move faster than waiting for a merge.

**Comment by FrostStarBook** [comment]
> It makes sense to maintain a fork of this rather than PR. This way the C&C client can move faster than waiting for a merge.

i see

**Pull request closed without merging**